tests/pipeline/test_column.py

```diff
@@ -81,3 +84,18 @@ def test_latest(self):
 
             expected_col_result = self.expected_latest(column, cal_slice)
             assert_frame_equal(col_result, expected_col_result)
+
+    @parameterized.expand([
+        (operator.gt,),
+        (operator.ge,),
+        (operator.lt,),
+        (operator.le,),
+    ])
+    def test_comparison_errors(self, compare):
+        columns = TDS.columns
+        for column in columns:
+            with self.assertRaises(BoundColumnInvalidCompare):
+                compare(column, 1000)
+
+            with self.assertRaises(BoundColumnInvalidCompare):
+                compare(column, 'test')
```

Comment:
These are good tests; but I think it's worth also testing the specific error message they produce. You can probably just replace `assertRaises` with `assertRaisesRegexp`: https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaisesRegexp

Testing the message will ensure that we know exactly how `other.__class__` gets rendered -- maybe `{other.__class__.__name__}` would be better? Maybe `{other.__class__} instance` would be more readable? You may have already experimented with variations in the REPL, but you can save future readers from having to do the same.

Testing the message also helps avoid a very common error, which I don't _think_ you've made here, but a future editor of your code might add. Can you spot the bug below?

```python
msg = (
    "Can't compare {column!s} with {other.__class__}.  Did you mean"
    "to append .latest?"
)
```

Reply:
In the updated test I use `assertRaisesRegexp`, the only thing I don't explicitly check for is the column name because I was running into regular expression comparison issues with some of them.  But I do explicitly check for `other.__class__.__name__`, so we know exactly how it gets rendered.

Reply:
Oh, and bug below is "meanto" which I have certainly done many of times!